### ISSUE TITLE
Set() doesn't work with Generics

### ISSUE THREAD

**Issue by eyalhakim** [typeorm/typeorm#3671]
Set method now takes a deep partial of the repository entity. However, there is a tslint error, if the repository type is defined with a generic parameter.

Code below says it all.

**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[X] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
```
interface Deletable {
    deletedAt: Date;
}

export class Test<T extends Deletable> {
    protected async softRemoveById(type: any, id: number): Promise<void> {
        const connection = getConnectionManager().get();
        const repo = connection.getRepository<T>(type);
        return repo
            .createQueryBuilder('entity')
            .update()
            .set({ deletedAt: new Date() })
            .where("id = :id", { id: id })
            .execute();
    }
}
```

**Comment by Kononnable** [comment]
Looks like you forgot to describe the issue.

**Comment by eyalhakim** [comment]
@Kononnable added a description. Thanks

**Comment by hitallow** [comment]
I have same problem.

**Comment by OSA413** [comment]
The issue is still reproducible

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgcwKYwMIQHbdQYxmBwFkBDbMtKOAXzgDMoIQ4AiGATzFWhDYDcAKCHBsMVFAZl8qOABFUAG3RkARisRC4OuABNl6VHoCCMAFwKyE4bRGoAHpFhx8SsgGcPcACqoPMAA8PnCOEth63ooqMOoqAHxaunBgzBKExnCenNj4cB4QDDAASqggEABuqABCnACSegAUXDyWFJwANHDAepbYAK4gapIAlEnJuvg4Aa44eITE2HAAvCjoWLgERKQUVJKNIwB0aDAHwhOT0-BQqJArs5sLOMfopZAewDDQnMHxzdyoEbnC5wG4wfpQJY3SDaEHJQ74G7WVAARX6kk41X6wCUBigjQA5KhxJ9OASRrC4TpDv0wHpkQdKVTDh50I0kAYYsYzH1UAB3KwSA50ClU3SHPkAC0kqEabB693MPTYXSQPUsCtoorFcEOjgI-SFQKZdloQA

**Comment by OSA413** [comment]
After some conversation over TypeScript Discord server, we've come up with the solution that you have to either delete the `<T>` in `const repo = connection.getRepository<T>(type);` or make it `<Deletable>` since you know that the type comes from such interface.

TLDR: one of two solutions:
1. Remove <T>: `const repo = connection.getRepository(type);`
2. Make it <Deletable>: `const repo = connection.getRepository<Deletable>(type);`
